goa v2: type *xxxResponseBody has no field or method Validate

The following error happens when trying to compile the generated code. (go run cmd/controllersvc/main.go is enough)
`gen/http/topology/server/types.go:117:27: body.Template.Validate undefined (type *TemplateResponseBody has no field or method Validate)`
`gen/http/topology/server/types.go:127:27: body.Template.Validate undefined (type *Template has no field or method Validate)`

I have managed to narrow the repro design to the following:
```
package design

import (
	. "goa.design/goa/http/design"
	. "goa.design/goa/http/dsl"
)

var _ = API("controller", func() {
	Title("sVirt Controller")
	Server("http://localhost:8080")
})

var _ = Service("topology", func() {
	HTTP(func() {
		Path("/topology")
	})

	Method("list", func() {
		Result(CollectionOf(StoredTopology))
		HTTP(func() {
			GET("/")
			Response(StatusOK)
		})
	})

	Method("show", func() {
		Payload(func() {
			Attribute("id", String)
			Required("id")
		})
		Result(StoredTopology)
		HTTP(func() {
			GET("/{id}")
			Response(StatusOK)
		})
	})
})

var StoredTopology = ResultType("application/vnd.svirt.controller.topology", func() {
	Reference(Topology)
	TypeName("StoredTopology")

	Attributes(func() {
		Attribute("id", String)
		Attribute("nodes")
	})

	View("default", func() {
		Attribute("id")

	})

	View("tiny", func() {
		Attribute("id")
	})

	Required("id")
})

var Topology = Type("Topology", func() {
	Attribute("nodes", ArrayOf(Node))
})

var Node = Type("Node", func() {
	Attribute("template", Template)
})

var Template = Type("Template", func() {
	Attribute("name", String)
	Attribute("version", String)
	Attribute("image", String)

	Required("name", "image")
})

var NotFound = Type("NotFound", func() {
	Attribute("message", String)
	Attribute("id", String)

	Required("message", "id")
})
```

--- a/codegen/validation.go
+++ b/codegen/validation.go
@@ -345,16 +345,15 @@ func recurseAttribute(att, catt *design.AttributeExpr, n, target, context string
 				// where there is validation but no actual
 				// validation code: if the validation is a
 				// required validation that applies to
-				// attributes that cannot be nil or empty string
-				// i.e. primitive types other than string.
+				// attributes that cannot be nil i.e. primitive
+				// types.
 				if !a.Validation.HasRequiredOnly() {
 					hasValidations = true
 					return done
 				}
 				for _, name := range a.Validation.Required {
 					att := design.AsObject(a.Type).Attribute(name)
-					if att != nil &&
-						(!design.IsPrimitive(att.Type) || att.Type.Kind() == design.StringKind) {
+					if att != nil && !design.IsPrimitive(att.Type) {
 						hasValidations = true
 						return done
 					}

